IPython/utils/io.py

```diff
@@ -173,6 +173,7 @@ def ask_yes_no(prompt, default=None, interrupt=None):
         except KeyboardInterrupt:
             if interrupt:
                 ans = interrupt
+                print("\r")
```

Comment:
I think this should be outside the if clause, i.e. indented one level less, so that it also does the right thing if the caller hasn't supplied an answer to use on interrupt.